core/src/main/java/org/infinispan/remoting/transport/Transport.java

```diff
@@ -146,6 +146,15 @@
     */
    int getViewId();
 
+   /**
+    * @return A {@link CompletableFuture} that completes when the transport has installed the expected view.
+    */
+   CompletableFuture<Void> viewFuture(int expectedViewId);
```

Comment:
Not sure I like the shunting of the "type" into the method name. Just view() would be enough.